chore(agw): Move `make format_all` into a separate bash script

MoritzThomasHuebner opened the pull request:
Closes #14904 
<!--
    Tag your PR title with the components that it touches along with
    the type of change
    E.g. "fix(orc8r): Fix reindexer race condition" or "feat(agw) ..."
-->

## Summary

- [x] Moved the make action into a separate bash script.
- [x] Introduced a loop to decrease code redundancy.
- [x] Updated a relevant readme.
- [x] Communicate the change with the community.
<!-- Enumerate changes you made and why you made them -->

## Test Plan

- [x] Executing the script locally fixes formatting errors in the three defined folders.

MoritzThomasHuebner committed:
chore(agw): Move `make format_all` into a separate bash script

Also updated a docstring

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

nstng commented on lte/gateway/format_all.sh:
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+# Copyright 2021 The Magma Authors.
```
2023

MoritzThomasHuebner replied:
:+1:

nstng commented on lte/gateway/format_all.sh:
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
couple of notes that are more or less NITs:
* name: this script is only for c code, format_all does not reflect that
* is there a better folder for this script - dev-tools maybe?
* this script fails if clang format is not installed - case on my host, but it's installed in the mgma vm
  * should we have feed-back in this case?
* the script takes some time, but is very silent
  * should we have some start, finish and current folder logging (or even current file)?
* using `set -euo pipefail` is a good pattern

MoritzThomasHuebner replied:
Replying point by point:
- Unsusre, I think we could name it `format_c.sh` perhaps
- I was thinking about that, but I did not find a good option. `dev_tools.py` is file in the directory `format_all.sh` currently is in. The only other `dev_tools` folder is in `cwf`, which would be out of place.
- I added a check if clang is installed and the script now exits on a helpful message.
- I changed it so it prints out every file
- I added the pipefail line

MoritzThomasHuebner committed:
chore(agw): Added pipefail and fixed year in license

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

MoritzThomasHuebner committed:
chore(agw): Added feedback if clang is not installed

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

MoritzThomasHuebner committed:
chore(agw): Print out every file that is formatted

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

MoritzThomasHuebner committed:
chore(agw): Updated moved clang script and updated documentation

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

LKreutzer commented on dev_tools/clang_format.sh:
```diff
@@ -0,0 +1,34 @@
+
+# This script formats all C/C++ files in the magma repo using clang-format-11.
+# The script can be executed on the dev VM or on the
+# host machine if clang-format-11 is installed.
```
Careful, if on the host `MAGMA_ROOT` won't be defined. It might be best to not support this for now.

LKreutzer replied:
Or you can ask for the user to set the variable correctly.

voisey replied:
Regarding "on the host VM", do you mean the host?

LKreutzer replied:
Sorry - of course - edited.

MoritzThomasHuebner replied:
Done

LKreutzer commented on dev_tools/clang_format.sh:
```diff
@@ -0,0 +1,34 @@
+
+for dir in /orc8r/gateway/c/ /lte/gateway/c/ /lte/gateway/python/;
+do
+  FILES=$(find "${MAGMA_ROOT}/${dir}" \( -iname "*.c" -o -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" \) -print)
```
The slash between `${MAGMA_ROOT}/${dir}` is too much, the `dir` already start with a slash`.

voisey replied:
Imo it makes it easier to parse as a path when reading the code though.

LKreutzer replied:
There are two options: either remove the slash here or in the list in line 26 at the beginning of each path. Double slashes work on Linux usually, but I would not want to keep it this way.

voisey replied:
Yes, I would favour the latter option.

MoritzThomasHuebner replied:
Done, removed the slash in L26

MoritzThomasHuebner committed:
chore(agw): Improved script

- Added check for if the MAGMA_ROOT variable is set
- Removed an unnecessary "/"

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>
